internal/authorization/rbac/rbac.go

```diff
@@ -173,3 +175,13 @@ func nonResourceURLMatches(rule *rbacv1.PolicyRule, requestedURL string) bool {
 	}
 	return false
 }
+
+// compareSubjects determines whether subjects names are equal (string equality is used).
+// If caseInsensitive is true, the case of the characters is ignored, meaning "UserName"
+// would be considered equal to "username".
+func compareSubjects(s1, s2 string, caseInsensitive bool) bool {
+	if caseInsensitive == false {
+		return s1 == s2
+	}
+	return strings.EqualFold(s1, s2)
```

Comment:
Is there any security ramifications of using EqualFold? This does more than just case-insensitive comparisons? cc @mhrabovcin